-1

Tested at: 815c34c6a617a0764449d977d8aa6cf8e8414727

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17179/17298/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>@@@@ Running edmWriteConfigs for SimG4CoreGFlashPlugins
--- Registered EDM Plugin: SimG4CoreGFlashPlugins
>> Leaving Package SimG4Core/GFlash
>> Package SimG4Core/GFlash built
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-15-2300/src/SimG4Core/GeometryProducer/src/GeometryProducer.cc: In member function 'void GeometryProducer::updateMagneticField(const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-15-2300/src/SimG4Core/GeometryProducer/src/GeometryProducer.cc:94:80: error: no matching function for call to 'sim::FieldBuilder::build(G4FieldManager*, G4PropagatorInField*)'
         m_fieldBuilder->build( tM->GetFieldManager(),tM->GetPropagatorInField());
                                                                                ^
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-15-2300/src/SimG4Core/GeometryProducer/src/GeometryProducer.cc:15:0:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_0_X_2017-01-15-2300/src/SimG4Core/MagneticField/interface/FieldBuilder.h:23:10: note: candidate: void sim::FieldBuilder::build(CMSFieldManager*, G4PropagatorInField*)
     void build(CMSFieldManager* fM, G4PropagatorInField* fP);
</pre>

--- a/SimG4Core/Application/python/g4SimHits_cfi.py
+++ b/SimG4Core/Application/python/g4SimHits_cfi.py
@@ -62,7 +62,7 @@ g4SimHits = cms.EDProducer("OscarMTProducer",
         NodeNames = cms.vstring('World')
     ),
     G4Commands = cms.vstring(),
-    SteppingVerbosity = cms.int32(1),
+    SteppingVerbosity = cms.int32(0),
     StepVerboseThreshold = cms.double(0.1), # in GeV
     VerboseEvents = cms.vint32(),
     VertexNumber  = cms.vint32(),

--- a/SimG4Core/GeometryProducer/src/GeometryProducer.cc
+++ b/SimG4Core/GeometryProducer/src/GeometryProducer.cc
@@ -13,6 +13,7 @@
 #include "SimG4Core/Geometry/interface/G4LogicalVolumeToDDLogicalPartMap.h"
 #include "SimG4Core/Geometry/interface/SensitiveDetectorCatalog.h"
 #include "SimG4Core/MagneticField/interface/FieldBuilder.h"
+#include "SimG4Core/MagneticField/interface/CMSFieldManager.h"
 #include "SimG4Core/MagneticField/interface/Field.h"
 #include "SimG4Core/Application/interface/SimTrackManager.h"
 
@@ -42,7 +43,7 @@ void createWatchers(const edm::ParameterSet& iP, SimActivityRegistry& iReg,
     for(std::vector<ParameterSet>::iterator itWatcher = watchers.begin();
 	itWatcher != watchers.end(); ++itWatcher) 
     {
-	std::auto_ptr<SimWatcherMakerBase> 
+	std::unique_ptr<SimWatcherMakerBase> 
 	    maker(SimWatcherFactory::get()->create(itWatcher->getParameter<std::string> ("type")));
 	if(maker.get()==nullptr) { 
 	  throw cms::Exception("SimG4CoreGeometryProducer", 
@@ -87,19 +88,19 @@ void GeometryProducer::updateMagneticField( edm::EventSetup const& es) {
        const GlobalPoint g(0.,0.,0.);
        edm::LogInfo("GeometryProducer") << "B-field(T) at (0,0,0)(cm): " << pMF->inTesla(g);
 
-       m_fieldBuilder = std::auto_ptr<sim::FieldBuilder>(new sim::FieldBuilder(&(*pMF), m_pField));
-
-           G4TransportationManager * tM = G4TransportationManager::GetTransportationManager();
-        // update field here ...          
-        m_fieldBuilder->build( tM->GetFieldManager(),tM->GetPropagatorInField());
-
-        edm::LogInfo("GeometryProducer") << "Magentic field updated";
+       sim::FieldBuilder fieldBuilder(pMF.product(), m_pField);
+       CMSFieldManager* fieldManager = new CMSFieldManager();
+       G4TransportationManager * tM = G4TransportationManager::GetTransportationManager();
+       tM->SetFieldManager(fieldManager);
+       fieldBuilder.build( fieldManager, tM->GetPropagatorInField());
+       edm::LogInfo("GeometryProducer") << "Magentic field is built";
     }
 }
  
-void GeometryProducer::beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const& es) {
-       // mag field can change in new lumi section
-       updateMagneticField( es );
+void GeometryProducer::beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&) 
+{
+  // mag field cannot be change in new lumi section - this is commented out
+  //     updateMagneticField( es );
 }
 
 void GeometryProducer::beginRun(const edm::Run &, const edm::EventSetup&)

